pkg/metadata/host/host.go

```diff
@@ -95,6 +97,24 @@ func getHostTags() *tags {
 	}
 }
 
+func getEC2Tags() ([]string, error) {
+	ec2Tags, err := ec2.GetTags()
+	if err != nil {
+		return ec2Tags, err
+	}
+
+	if !config.Datadog.GetBool("collect_security_groups") {
+		// remove the `security_group_name` tag if present
+		for i, s := range ec2Tags {
+			if strings.HasPrefix(s, "security_group_name:") {
```

Comment:
@arbll if you look at the A5 logic: https://github.com/DataDog/dd-agent/blob/5.21.2/utils/cloud_metadata.py#L212-L214 it looks like the security_group_name tag is not part of the tags that are returned by the AWS API, but that we extract the tag from the instance metadata and append it to the list of tags. So we should probably do the same thing in A6 so that users who do have the collect_security_groups option enabled have the tag with A6.

What do you think?